cylc/uiserver/data_store_mgr.py

```diff
@@ -144,13 +143,23 @@ def update_workflow_data(self, topic, delta, w_id):
             delta, 'time', getattr(delta, 'last_updated', 0.0))
         # If the workflow has reloaded recreate the data
         # otherwise apply the delta if it's newer than the previously applied.
-        if delta.reloaded:
-            self.data[w_id][topic] = {ele.id: ele for ele in delta.deltas}
-            self.data[w_id]['delta_times'][topic] = delta_time
-        elif delta_time >= self.data[w_id]['delta_times'][topic]:
-            apply_delta(topic, delta, self.data[w_id])
-            self.data[w_id]['delta_times'][topic] = delta_time
-            self.reconcile_update(topic, delta, w_id)
+        try:
+            if delta.reloaded:
+                if hasattr(delta, 'last_updated'):
+                    self.data[w_id][topic].CopyFrom(delta)
+                else:
+                    self.data[w_id][topic] = {
+                        ele.id: ele
+                        for ele in delta.deltas
+                    }
+                self.data[w_id]['delta_times'][topic] = delta_time
+            elif delta_time >= self.data[w_id]['delta_times'][topic]:
+                apply_delta(topic, delta, self.data[w_id])
+                self.data[w_id]['delta_times'][topic] = delta_time
+                self.reconcile_update(topic, delta, w_id)
+        except Exception:
```

Comment:
Can/should we catch `KeyError` or some other more specific exception?

Reply:
I guess a lot of things could potentially go wrong in the application and reconciliation of the UIS data-store sync (attribute errors should be the most rare, and only happen with API changes).

Reply:
Thanks for confirming! Marking as resolved then.

Reply:
> I guess a lot of things could potentially go wrong in the application and reconciliation of deltas in the UIS data-store sync

The usual practice is to handle specific known exceptions, and crash spectacularly if something new and unknown goes wrong (which is how you identify what else can go wrong that you haven't thought of, rather than hiding the problem).  Is there some reason to do it differently here?

Reply:
> > I guess a lot of things could potentially go wrong in the application and reconciliation of deltas in the UIS data-store sync
> 
> The usual practice is to handle specific known exceptions, and crash spectacularly if something new and unknown goes wrong (which is how you identify what else can go wrong that you haven't thought of, rather than hiding the problem). Is there some reason to do it differently here?

The issue is it won't crash the program because it runs in a thread pool (we don't even see the error)..

Reply:
@hjoliver - ok removed now.